data/scripts.js

```diff
@@ -3644,9 +3644,20 @@ exports.BattleScripts = {
 		var pool = Object.keys(sets).randomize();
 		var ranks = {'~':'admins', '&':'leaders', '@':'mods', '%':'drivers', '+':'voices'};
 		var levels = {'~':99, '&':97, '@':96, '%':96, '+':95};
-		for (var i = 0; i < 6; i++) {
-			var rank = pool[i].charAt(0);
+		var count = 0;
+		var i = -1;
+		do {
+			i++;
 			var set = sets[pool[i]];
+			var rejected = false;
+			for (var otherSet in team) {
+				if (otherSet.species === set.species) {
```

Comment:
Will this ever be `true`? I don't think so, as there are no duplicate keys.
So, this whole check becomes redundant. (and hence, `count` and `i` are equal).
@Joimer